haystack/document_store/sql.py

```diff
@@ -80,7 +87,11 @@ def __init__(
         :param update_existing_documents: Whether to update any existing documents with the same ID when adding
                                           documents. When set as True, any document with an existing ID gets updated.
                                           If set to False, an error is raised if the document ID of the document being
-                                          added already exists. Using this parameter coud cause performance degradation for document insertion. 
+                                          added already exists. Using this parameter could cause performance degradation
+                                          for document insertion.
+        :param batch_size: Maximum number of host parameters in a single SQL statement.
```

Comment:
Let's make this docstring a bit clearer for users. In the implementation, it seems that `batch_size` is usually related to the number of rows (i.e. usually docs) that we query in a single statement. Is that really the same as the number of host parameters or can there be >1 host params per row? We should also state what the effect of a small value is and decide what a good default is. From my understanding, "999" is the safe bet to support older SQLite versions. How much will this default value slow down larger production workloads (e.g. Postgresql with 500k docs)? If this slowdown is significant we should rather increase the default here and only switch it to 999 for  sqlite < 3.32.0

Reply:
Yeah you are correct about misleading docstring.
Actually it is not related to `SQLITE_MAX_VARIABLE_NUMBER`, but solve this issue as well.
Mainly it is solving following issues -
- Limit is number of variables because SQLite  allocates space to hold all host parameters
- Limit the memory usage to hold rows.objects in the RAM

Main performance boost is coming by using two select instead of ORM it solve two issues -
- ORM build objects in memory use cache, and hence need lot of memory.
- ORM need to perform more network calls then plain SQL (specially helpful in case of Postgres or MySQL)
- Refer https://stackoverflow.com/questions/494816/using-an-orm-or-plain-sql

I think better to keep `batch_size` higher means like `32766` which is default for version after `3.32.0`.  But selection of this value depend on the host machine main memory. But I am not sure about good value for production workload until we have good benchmarking. I think we would need to spend some time in it.

Reply:
I updated this doc string please check.
I wanted to write this `Higher batch_size value is good for performance but consume more main memory`. But did not do it as we dont have benchmarks to prove it.

Reply:
@tholor could you please update doc string if current comment need some improvement.